wireapp/wire-ios-mono
# feat: start mls conference WPB-613

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-613" title="WPB-613" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-613</a>  [iOS] Start an MLS conference call
  </td></table>
  <br />
 










----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

In this PR we add support for starting a conference call in an MLS group. Changes include:

- Adding some missing mls action handlers to fix mls requests not being made to the BE
- Fix some decoding of Subgroup payloads
- Storing sub conversation group ids so we can fetch them later
- Adding support for decoding subconversation type in update events
- When decrypting a message, use the subconversation group id if needed
- When starting a call, if it's an mls conference, join the subconversation, generate conference info, pass conference info to AVS
- Updating mocks, 
- Cleaning up some extensions to reduce duplication and increase discovery
- General grooming

### Testing

#### Test Coverage

- Unit tests asserting subconversation group id repository
- Unit tests asserting we store subconversation group id
- Unit tests asserting we decrypt messages for subconversation
- Unit test we start mls conferences
- Updating existing tests for API changes

#### How to Test

- Create an MLS group.
- Start a call.
- Answer call from a remote side (e.g Chrome)
- Verify the call connects.

### Notes
Although the call will connect, the call may not be entirely functional yet due to some missing pieces which we'll address in other PRs.

There's a lot of duplication with mocks which I want to finally address in a separate PR.

Currently we will decrypt a message for a subconversation if needed in the main app, but we don't do that in the notification extension at the moment. I'm not sure if we need too, if we would ever get a message in a subconversation while the app is in the background. If we do, then we need to find a way to persist the subconversation group id because it's currently only held in memory.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
